test/e2e/framework/cluster.go

```diff
@@ -109,13 +109,17 @@ func (tf *TestFramework) WaitForClusterCondition(aerospikeCluster *aerospikev1al
 	return nil
 }
 
-func (tf *TestFramework) WaitForClusterNodeCount(aerospikeCluster *aerospikev1alpha2.AerospikeCluster, nodeCount int32) error {
+func (tf *TestFramework) WaitForClusterNodeCountOrTimeout(aerospikeCluster *aerospikev1alpha2.AerospikeCluster, nodeCount int32, timeout time.Duration) error {
 	return tf.WaitForClusterCondition(aerospikeCluster, func(event watchapi.Event) (bool, error) {
 		// grab the current cluster object from the event
 		obj := event.Object.(*aerospikev1alpha2.AerospikeCluster)
 		// search for the current node count
 		return obj.Status.NodeCount == nodeCount, nil
-	}, watchTimeout)
+	}, timeout)
+}
+
+func (tf *TestFramework) WaitForClusterNodeCount(aerospikeCluster *aerospikev1alpha2.AerospikeCluster, nodeCount int32) error {
+	return tf.WaitForClusterNodeCountOrTimeout(aerospikeCluster, nodeCount, watchTimeout)
```

Comment:
I don't understand the changes in this file. `WaitForClusterNodeCount` already returns an error when `watchTimeout` is crossed. Now, your changes add a parameterized `timeout` and then you pass `watchTimeout`, which seems another layer of indirection that is only contributing to more LOC and decreased readability. At most, I'd be OK with renaming `WaitForClusterNodeCount` to `WaitForClusterNodeCountOrTimeout` but I really don't think it will add any value.

Reply:
For one of the tests I am sending a node selector that doesn't match any node and waiting for it to fail. I created this method to allow to setup a specific timeout time instead of relying on default one which is really big and will end up slowing down the tests a lot. Does it make sense?

Reply:
Yes, I had missed the test that calls said method. Thanks for clarifying.